sound/soc/sof/intel/hda-dsp.c

```diff
@@ -297,9 +297,7 @@ static int hda_suspend(struct snd_sof_dev *sdev, bool runtime_suspend)
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 	if (IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC) && runtime_suspend)
 		/* enable controller wake up event */
-		snd_hdac_chip_writew(bus, WAKEEN,
-				     snd_hdac_chip_readw(bus, WAKEEN) |
-				     hda->hda_codec_mask);
+		snd_hdac_chip_updatew(bus, WAKEEN, 0, hda->hda_codec_mask);
```

Comment:
Hmm, the macros in hdaudio.h are a bit odd in that the mask is not applied to the value, so this still works, but it looks odd. Can you fix this to

snd_hdac_chip_updatew(bus, WAKEEN, hda->hda_codec_mask, hda->hda_codec_mask);

... just in case someone refactors the hdaudio.h macros (to behave like e.g. snd_sof_dsp_update_bits() family works).

Reply:
@kv2019i I'm not sure snd_sof_dsp_update_bits() is a common way to handle bitmasks, cf. set_mask_bits()

Reply:
@lyakh True, but in ALSA we seem to be using the sof_dsp_update_bits style (see e.g. use of snd_hdac_chip_updateb in hda_controller.c). Also regmap_update_bits applies the mask to value.